lib/HLSL/HLSignatureLower.cpp

```diff
@@ -1159,21 +1159,23 @@ void HLSignatureLower::GenerateDxilCSInputs() {
     }
 
     Constant *OpArg = hlslOP->GetU32Const((unsigned)opcode);
-    Type *Ty = arg.getType();
-    if (Ty->isPointerTy())
-      Ty = Ty->getPointerElementType();
-    Function *dxilFunc = hlslOP->GetOpFunc(opcode, Ty->getScalarType());
+    Type *NumTy = arg.getType();
+    if (NumTy->isPointerTy())
+      NumTy = NumTy->getPointerElementType();
```

Comment:
Can pointer be valid for any of these?  I don't think so...

Reply:
Changed to asserts and filed #2299